docs/extensions/payment-processors/create.md

```diff
@@ -1,5 +1,227 @@
 
+This page explains how to develop a payment processor extension.
+
+## Intro
+
+A payment processor integration extension typically includes:
+
+1. a Payment Processor class (required) - this provides CiviCRM with a standard interface/API bridge to the third party processor, and a way for CiviCRM to know which features are supported. This must extend `CRM_Core_Payment`.
+
+2. an IPN class - optional, IPN means *Instant Payment Notification*, although they are usually asynchronous and not "instant". Many third parties talk instead about *Webhooks*.
+
+    It refers to the data sent from the third party on various events e.g.:
+
+    - completed/confirmed payment,
+    - cancellation of recurring payment,
+    - often many more situations - depends heavily on the third party, often configurable in their account administration facilities
+
+    CiviCRM provides a base class for IPN classes `CRM_Core_Payment_BaseIPN`, and a menu route at `civicrm/payment/ipn/<N>` where `<N>` is the payment processor ID.
+
+3. Customisations and additions to the settings. e.g. might use an private *API key instead* of a *password*.
+
+4. Other libraries and helpers for the particular Payment Processor service.
+
+
+## The Payment Class
+
+A payment processor object extends `CRM_Core_Payment`. This class provides CiviCRM with a standard interface/API bridge to the third party processor. It should be found at:
+
+```
+<myextension>/CRM/Core/Payment/MyExtension.php
+```
+
+The class handles data to do with the third party processor's needs. Different methods will require different data from CiviCRM, e.g. from Contribution or ContributionRecur records, and configuration data for the Payment Processor Service.
+
+!!!important
+    Try to avoid infringing on CiviCRM's logic. The methods in your extension should take inputs, communicate with the third party, and return output data that CiviCRM can use to perform its logic. If you find your extension is sending emails, duplicating logic, updating or creating records in CiviCRM, outputting user content (e.g. status messages) then stop, check and consider separating out your code into different methods.
+
+Because things get complex and because `CRM_Core_Payment` is a bridge between CiviCRM's logic and the payment processor service's needs, it's all too easy to end up combining business logic (like updating membership end dates, or deciding whether to send a receipt email) with your calls to the external service. If you do need to do this try to separate out your class methods.
+
+CiviCRM's Contribution and Event pages are obvious users of your `CRM_Core_Paymnet` class but you should not assume that those are the only consumers; it should be able to be used by other processes too, e.g. Drupal webform or an entirely bespoke process. **Therefore they should not call functions that assume a user context** such as redirects, exits, or setting status messages like `CRM_Core_Error::statusBounce`.
```

Comment:
I'm highly concerned about any approach that suggests regularly overriding a core file of any kind, but especially something as complex and touchy as CRM_Core_Payment. I'd prefer to recommend that before any payment process tries overriding that file that they discuss their needs with the community to try to figure out a way to make core serve their needs.

Reply:
Sorry, it's much clearer now that I read it over again, saying it extends rather than overrides. Maybe mention implementing stub classes or abstract interfaces? There should not be interactions with business logic eg for handling memberships..that's a clear case of a problem. If necessary core needs changes but more likely new developer needs to be shown right way or workaround way.